tests/scripts/run_benchmarks.sh

```diff
@@ -39,33 +48,45 @@ done
 cat benchmark_${LASTCOMMIT}.csv
 
 echo "===> Install components for benchmark analysis ..."
+
 pip3 install matplotlib numpy pandas
 
 echo "===> Download benchmark history ..."
 
-cd ${GITHUB_WORKSPACE}
-
+# Configure Github user
 git config --local user.email "action@github.com"
 git config --local user.name "GitHub Action"
 
-git fetch --no-recurse-submodules
-cd extern/algebra-plugins
-git reset --hard origin/main
-cd -
-git submodule update --init --recursive
-
+# Checkout to benchmark branch
 git checkout -b gh-pages origin/gh-pages
-cp archive/benchmarks/benchmarks_history.csv ${PWD_BUILD}/.
-cd ${PWD_BUILD}
-cat benchmark_${LASTCOMMIT}.csv >> benchmarks_history.csv
+git pull origin gh-pages
+git reset --hard origin/gh-pages
+
+git checkout ${LASTCOMMIT} -- ${WORKSPACE}/tests/scripts/analyze_benchmarks.py 
+
+# Record the benchmark result into benchmarks_history 
+BENCHMARK_HISTORY_FILE=${WORKSPACE}/archive/benchmarks/benchmarks_history.csv
+cat ${WORKSPACE}/benchmark_${LASTCOMMIT}.csv >> ${BENCHMARK_HISTORY_FILE}
 
 echo "===> Running benchmark analysis ..."
-python3 -i analyze_benchmarks.py
-
-echo "===> Prepare for uploading results ..."
-cp benchmarks_history.csv ${GITHUB_WORKSPACE}/archive/benchmarks/benchmarks_history.csv
-cp *.png  ${GITHUB_WORKSPACE}/figures/.
-cd ${GITHUB_WORKSPACE}
-git add archive/benchmarks/benchmarks_history.csv
-git add figures/*.png
-git commit -m"updating benchmark data for commit ${LASTCOMMIT}" -a
+
+cd ${WORKSPACE}
+python3 ${BASEDIR}/analyze_benchmarks.py ${BENCHMARK_HISTORY_FILE}
+
+echo "===> Commiting the results ..."
+
+cp ${WORKSPACE}/*.png  ${WORKSPACE}/figures/.
+
+git add ${BENCHMARK_HISTORY_FILE}
+git add ${WORKSPACE}/figures/*.png
+
+git commit -m "updating benchmark data for commit ${LASTCOMMIT}" -a
+
+if [ -z "${GITHUB_WORKSPACE}" ]; then
+    echo "===> Exit"
+else
+    echo "===> Uploading the results"
+    # CI stuck here...
+    #git push --verbose origin gh-pages
```

Comment:
Of course `git push` doesn't work without a valide token.. let's find another way. As Andi mentioned, we can benchmark what Paul did for ACTS main repository

Reply:
Yes, indeed - the project was also shifted from my private to the public acts repo at some time, not sure if this was handled transparently.